build/building/context.md

```diff
@@ -150,7 +158,36 @@ RUN --mount=target=. \
 ```
 
 ```console
-$ docker build --build-arg BUILDKIT_CONTEXT_KEEP_GIT_DIR=1 https://github.com/user/myrepo.git#main
+$ docker build \
+  --build-arg BUILDKIT_CONTEXT_KEEP_GIT_DIR=1
+  https://github.com/user/myrepo.git#main
+```
+
+#### Private repositories
+
+When you use a Git repository as the build context, the Buildx client
+downloads the repository to a temporary directory on its host before
+sending it to the BuildKit daemon.
+
+Downloading the repository to the client host system first lets you
+access private repositories using things like local user credentials
+and VPN configurations that exists on the host machine.
+
+You can also use token-based authentication with Git contexts. Pass
+the token to the build using the `--secret` flag.
+
+```console
+$ GIT_AUTH_TOKEN=<token> docker buildx build \
+  --secret id=GIT_AUTH_TOKEN \
+  https://github.com/user/private.git
+```
+
+Buildx automatically detects and uses SSH credentials if the specified
```

Comment:
Since this is the default, I think we could put this before the token-based authentication?

Reply:
Good point! I'll move it